Source/playerdat.cpp

```diff
@@ -152,11 +153,99 @@ void ReloadExperienceData()
 	}
 }
 
+void LoadClassAttributes(std::string_view classPath, ClassAttributes &out)
+{
+	const std::string filename = StrCat("txtdata\\classes\\", classPath, "\\attributes.tsv");
+	tl::expected<DataFile, DataFile::Error> dataFileResult = DataFile::load(filename);
+	if (!dataFileResult.has_value()) {
+		DataFile::reportFatalError(dataFileResult.error(), filename);
+	}
+	DataFile &dataFile = dataFileResult.value();
+
+	if (tl::expected<void, DataFile::Error> result = dataFile.skipHeader();
+	    !result.has_value()) {
+		DataFile::reportFatalError(result.error(), filename);
+	}
+
+	auto recordIt = dataFile.begin();
+	const auto recordEnd = dataFile.end();
+
+	const auto getValueField = [&](std::string_view expectedKey) {
+		if (recordIt == recordEnd) {
+			app_fatal(fmt::format("Missing field {} in {}", expectedKey, filename));
+		}
+		DataFileRecord record = *recordIt;
+		FieldIterator fieldIt = record.begin();
+		const FieldIterator endField = record.end();
+
+		const std::string_view key = (*fieldIt).value();
+		if (key != expectedKey) {
+			app_fatal(fmt::format("Unexpected field in {}: got {}, expected {}", filename, key, expectedKey));
+		}
+
+		++fieldIt;
+		if (fieldIt == endField) {
+			DataFile::reportFatalError(DataFile::Error::NotEnoughColumns, filename);
+		}
+		return *fieldIt;
+	};
+
+	const auto readRecord = [&](std::string_view expectedKey, auto &outValue) {
+		DataFileField valueField = getValueField(expectedKey);
+		tl::expected<void, devilution::DataFileField::Error> parseIntResult;
+		if constexpr (std::is_same_v<std::remove_cvref_t<decltype(outValue)>, int16_t>) {

```

Comment:
Is it possible to use this with a typedef for a fixed_16_6 type?  to prevent any surprises if someone changes one of the stat fields to int16_t for example

Reply:
Done